components/reports/QueueFilter/Tag.tsx

```diff
@@ -0,0 +1,220 @@
+import { XCircleIcon } from '@heroicons/react/24/solid'
+import { classNames } from '@/lib/util'
+import { useQueueFilter } from '../useQueueFilter'
+import { ActionButton } from '@/common/buttons'
+import Select from 'react-tailwindcss-select'
+
+const availableTagOptions = {
+  report: {
+    text: 'Report Type',
+    options: [
+      { text: 'Spam', value: 'report:spam' },
+      { text: 'Rude', value: 'report:rude' },
+      { text: 'Other', value: 'report:other' },
+      { text: 'Violation', value: 'report:violation' },
+      { text: 'Misleading', value: 'report:misleading' },
+    ],
+  },
+  embed: {
+    text: 'Embedded Content',
+    options: [
+      { text: 'Image', value: 'embed:image' },
+      { text: 'Video', value: 'embed:video' },
+      { text: 'Link/External', value: 'embed:external' },
+    ],
+  },
+  lang: {
+    text: 'Language',
+    options: [
+      { text: 'English', value: 'lang:en' },
+      { text: 'Portuguese', value: 'lang:pt' },
+      { text: 'Spanish', value: 'lang:es' },
+      { text: 'French', value: 'lang:fr' },
+      { text: 'Japanese', value: 'lang:ja' },
+      { text: 'German', value: 'lang:de' },
+      { text: 'Italian', value: 'lang:it' },
+      { text: 'Korean', value: 'lang:ko' },
+      { text: 'Russian', value: 'lang:ru' },
+      { text: 'Chinese', value: 'lang:zh' },
+      { text: 'Arabic', value: 'lang:ar' },
+      { text: 'Unknown', value: 'lang:und' },
+    ],
+  },
+}
+
+const getTagOptions = (subjectType?: string) => {
+  const { embed, ...rest } = availableTagOptions
+  return Object.values(subjectType === 'account' ? rest : availableTagOptions)
+}
+
+const selectClassNames = {
+  tagItemIconContainer:
```

Comment:
These classes are mostly copy pasted from the library itself because the lib doesn't append/overwrite classes and instead, replaces entirely based on customized config so in order to maintain current styling AND add dark mode, we need to bring over all classes the lib uses and add our own dark mode classes.